### ISSUE TITLE
Included relation name, fields[TYPE] parameter and db table name assumed to be equal

### ISSUE THREAD

**Issue by arad0s** [spatie/laravel-query-builder#134]
According to the JSON:API specs, under the Sparse Fieldsets section, "A client MAY request that an endpoint return only specific fields in the response on a per-type basis by including a fields[TYPE] parameter", so the following request should return only the title and body of the articles, and only the name of the included author: 
GET /articles?include=**author**&fields[articles]=title,body&fields[**people**]=name HTTP/1.1

However, all of the included relation's fields are selected from the database (name, email, address...).

This happens because in the QueryBuilder@addIncludeToQuery method, it is assumed that the name of the included relation (author) is the same as:
1. The TYPE stated in the fields parameter (people).
2. The database table name of the included resource.

My solution for this issue was modifying the QueryBuilder@getFieldsForRelatedTable method from:
```
    protected function getFieldsForRelatedTable(string $relation): array
    {
        if (! $this->fields) {
            return ['*'];
        }

        return $this->fields->get($relation, []);
    }
```
to:
```
    protected function getFieldsForRelatedTable(string $relation): array
    {
        if (! $this->fields) {
            return ['*'];
        }

        return $this->fields->get($this->model->$relation()->getRelated()->getTable(), []);
    }
```
This change won't help in case the related type is different from the table name (e.g. the type is people but the table name is users), but it eliminates the assumption that the name of the included relation would be the same as the type stated in the fields parameter.

I will be glad if the above issue is fixed and will also like to hear a solution for the type != table name problem.

Thanks.

**Comment by freekmurze** [comment]
Thanks for the detailed info. Could you PR the above with tests?

**Comment by dominikb** [comment]
Would the table != type alias problem be solved with a solution as described in #137?

If so, I'd be down to implement both of those in one PR.

**Comment by arad0s** [comment]
I'm not quiet sure whether the solution described in #137 will solve the table != type problem.

As for my understanding, if the array given to the allowedIncludes method will be consisted of relation_name => resource_type pairs (e.g. 'author' => 'people') such that the resource_type is used as the value sent to the getFieldsForRelatedType method, the problem I described will be fixed, including the table != type problem (without changing the implementation of the getFieldsForRelatedType). But I don't think that's what #137 referred to.

Or maybe there is no good justification of having a resource who's table is different than its type. If so, my solution suffices and has nothing to do with #137.
